cfme/tests/services/test_ansible_workflow_servicecatalogs.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-:
+import pytest
+
+from cfme import test_requirements
+from cfme.services.myservice import MyService
+from cfme.services.service_catalogs import ServiceCatalogs
+from cfme.utils import testgen
+from cfme.utils.blockers import GH
+from cfme.utils.log import logger
+
+
+pytestmark = [
+    test_requirements.service,
+    pytest.mark.tier(2),
+    pytest.mark.ignore_stream('upstream')]
+
+
+def pytest_generate_tests(metafunc):
+    # Filter out providers without provisioning data or hosts defined
+    argnames, argvalues, idlist = testgen.config_managers(metafunc)
+    new_idlist = []
+    new_argvalues = []
+    for i, argvalue_tuple in enumerate(argvalues):
+        args = dict(zip(argnames, argvalue_tuple))
+
+        if not args['config_manager_obj'].yaml_data['provisioning']:
+            continue
+
+        new_idlist.append(idlist[i])
+        new_argvalues.append(argvalues[i])
+
+    testgen.parametrize(metafunc, argnames, new_argvalues, ids=new_idlist, scope='module')
+
+
+@pytest.fixture(scope="module")
+def config_manager(config_manager_obj):
+    """ Fixture that provides a random config manager and sets it up"""
+    if config_manager_obj.type == "Ansible Tower":
+        config_manager_obj.create(validate=True)
+    else:
+        config_manager_obj.create()
+    yield config_manager_obj
+    config_manager_obj.delete()
+
+
+@pytest.fixture(scope="function")
+def catalog_item(appliance, request, config_manager, dialog, catalog):
+    job_type = 'workflow'
+    config_manager_obj = config_manager
+    provider_name = config_manager_obj.yaml_data.get('name')
+    template = config_manager_obj.yaml_data['provisioning_data'][job_type]
```

Comment:
If this field is not present in the `yamls`, we should skip the test. What are your thoughts?

Reply:
Yeah, ideally if 'job_type'  is not present in the yamls, the test should be skipped, or the test shouldn't even be collected for the provider in question.

I'm already doing some filtering in pytest_generate_tests().
for i, argvalue_tuple in enumerate(argvalues):
        args = dict(zip(argnames, argvalue_tuple))

        if not args['config_manager_obj'].yaml_data['provisioning']:
            continue